enhancements/sig-policy/28-selective-policy-enforcment/README.md

```diff
@@ -0,0 +1,415 @@
+1. Give users visibility into the effect of a Configuration Policy
+   change prior to remediation
+1. Reduce complexity and scalability issues with alternative
+   approaches
+
+### Non-Goals
+
+Out of scope for this feature
+1. A mechanism/feature to capture/define the timing of enforcement
+1. A feature for managing progressive rollouts. The timing and cluster
+   selection are typically use-case specific. There may be enough
+   commonality to define a feature to manage progressive rollouts but
+   that is deferred to a separate discussion/enhancement.
+
+## Proposal
+
+The Selective Policy Enforcement feature adds an API to the policy
+framework to allow enforcement of inform policies to selected
+clusters. The effect of this would be the equivalent of setting the
+complianceType of child (cluster namespace copy) policies to enforce.
+
+The proposed API would be the addition of a remediationAction of
+“controlled” to Policy. In this mode the policy would act as an inform
+policy until the user, controller, or higher level orchestration
+enables remediation directly on the child policy.
+
+The mechanism to enable enforcement of a child policy would be to add
+an annotation to the child policy which enables enforcement
+
+```
+metadata:
+  annotations:
+    versioned-policy-enable: [any|<generation>]
+```
+
+The values of the annotation govern how the controller will handle
+modifications to the parent policy specification while the child
+policy is in enforcing mode.
+
+The values of the annotation can take either of these forms:
++ The value of `any` indicates that changes to the parent policy are
+immediately passed through to the enforcement of the child policy. As
+long as this annotation has the value of `any` remediation is enabled
+on the child policy and changes to the parent policy are immediately
+enforced to the cluster.
++ An integer `generation` value indicates that the child policy will
+continue enforcing as long as the child policy `metadata/generation`
+value is equal to the specified value. In this way the user of this
+feature has control over whether changes to the parent policy happen
+immediately or require additional intervention by the user. At the
+expense of this additional complexity the behavior closes race
+conditions on a user/controller enabling/disabling enforcement and
+near-concurrent changes to the parent policy specification.
+
+The overall pattern follows a similar one in OLM operator updates
+where a manual installPlanApproval is explicitly enabled by the user
+when an update is desired.
+
+The implementation is relatively simple and defers the “selection”
+flexibility to the user/orchestrator/controller
+
+To improve user visibility the dashboard view of policies should be
+enhanced to show the current enforcement state of the policy. This
+would be reflected in the Clusters tab of the Policies/_policyName_
+page and the specific child policy view
+Policies/_policyName_/_cluster-policy-name_
```

Comment:
This section isn't relevant for upstream since a UI hasn't been developed yet here.